compiler/src/iree/compiler/Codegen/Common/CPU/CPUPrepareUkernels.cpp

```diff
@@ -299,24 +301,31 @@ struct Convert3DPackto2DPackPattern : public OpRewritePattern<linalg::PackOp> {
     }
 
     Location loc = packOp.getLoc();
-    auto reducedSrcType =
-        RankedTensorType::Builder(packOp.getSourceType()).dropDim(srcPos);
-    auto reducedSrc = tensor::createCanonicalRankReducingExtractSliceOp(
-        rewriter, loc, packOp.getSource(), reducedSrcType);
-
-    auto reducedDestType =
-        RankedTensorType::Builder(packOp.getDestType()).dropDim(destPos);
-    auto reducedDest = tensor::createCanonicalRankReducingExtractSliceOp(
-        rewriter, loc, packOp.getDest(), reducedDestType);
-
-    auto newPackOp = rewriter.create<linalg::PackOp>(
-        loc, reducedSrc, reducedDest, newInnerDimsPos, packOp.getMixedTiles(),
-        packOp.getPaddingValue(), newOuterDimsPerm);
-
-    auto insertSliceOp = tensor::createCanonicalRankReducingInsertSliceOp(
-        rewriter, loc, newPackOp.getResult(), packOp.getDest());
-    rewriter.replaceOp(packOp, insertSliceOp);
-    return success();
+    auto srcType = packOp.getSourceType();
+    if (isa<RankedTensorType>(srcType)) {
+      RankedTensorType srcType = cast<RankedTensorType>(packOp.getSourceType());
+      RankedTensorType dstType = cast<RankedTensorType>(packOp.getDestType());
+      auto reducedSrcType =
+          RankedTensorType::Builder(srcType).dropDim(srcPos);
+      auto reducedSrc = tensor::createCanonicalRankReducingExtractSliceOp(
+          rewriter, loc, packOp.getSource(), reducedSrcType);
+
+      auto reducedDestType =
+          RankedTensorType::Builder(dstType).dropDim(destPos);
+      auto reducedDest = tensor::createCanonicalRankReducingExtractSliceOp(
+          rewriter, loc, packOp.getDest(), reducedDestType);
+      auto newPackOp = rewriter.create<linalg::PackOp>(
+            loc, reducedSrc, reducedDest, newInnerDimsPos, packOp.getMixedTiles(),
+            packOp.getPaddingValue(), newOuterDimsPerm);
+        auto insertSliceOp = tensor::createCanonicalRankReducingInsertSliceOp(
+            rewriter, loc, newPackOp.getResult(), packOp.getDest());
+        rewriter.replaceOp(packOp, insertSliceOp);
+        return success();
+    } else {
+      MemRefType srcType = cast<MemRefType>(srcType);
+      // TODO (ita9naiwa): add this case : add this case
```

Comment:
Follow the `// TODO(ita9naiwa): <an description of what is to be done>.` format for TODO messages.

TODOs added to the codebase should either be fully descriptive enough for someone else to come along and implement it with your name indicating you plan to do it ASAP or reference an issue instead (`// TODO(#1234): <description>.`. "add this case" is something you can use locally when developing but it's not something that belongs in the repository.